tools/rpm2img

```diff
@@ -257,11 +266,171 @@ fi
 # package has placed the image in /boot/efi/EFI/BOOT.
 mv "${ROOT_MOUNT}/boot/efi"/* "${EFI_MOUNT}"
 
+# For an AWS profile, we expect a config file for the PKCS11
+# helper. Otherwise, there should be a local key and cert.
```

Comment:
> To ensure I'm not missing anything (and it gets documented in the code if I am, to prevent future refactoring), the following blocks could be conditional on the use of Secure Boot, right?

I'll clean this up and move it all into a Secure Boot conditional. I'd wanted to ensure that the setup parts always got exercised, since we always expect a valid signing profile, but doing it this way means that it'd be easy to miss cases where we try to sign something or copy a file that should only exist when the feature flag is turned on.

> For example, shim could live with the all-zeros vendor_cert structure already embedded (unlike GRUB where the pubkey section needs to be removed)?

`shim` ignores the cert if Secure Boot is not enabled. However, in double-checking this I found a discrepancy in `mokutil` output:

```
# All-zeroes vendor cert specified at shim build time, and never replaced.
$ mokutil --list-enrolled
[key 1]
Invalid X509 certificate

# No vendor cert specified at shim build time.
]$ mokutil --list-enrolled
[key 1]
  [SHA-256]
  0000000000000000000000000000000000000000000000000000000000000000
```

Replacing the empty cert with a zero-length cert cleans this up so we always get the second case.